src/main/java/com/segment/analytics/android/integrations/amplitude/AmplitudeIntegration.java

```diff
@@ -264,11 +264,22 @@ public void track(TrackPayload track) {
   }
 
   private void event(
-      @NonNull String name, @NonNull Properties properties, @Nullable JSONObject groups) {
+      @NonNull String name,
+      @NonNull Properties properties,
+      @Nullable Map options,
+      @Nullable JSONObject groups) {
     JSONObject propertiesJSON = properties.toJsonObject();
-    amplitude.logEvent(name, propertiesJSON, groups);
+    boolean outOfSession = false;
+
+    if (!isNullOrEmpty(options)) {
```

Comment:
It might be a good improvement to make this a utility function, `getOptOutOfSessionFromOptions` that has  it's own tests for all the cases - map empty, contains outOfSession but null, does not contain outOfSession, contains outOfSession but not a boolean, etc.

Doesn't need to be handled in this PR though.

Reply:
Will do.